### PR TITLE
Replace trusty PPA build by static PPA build.

### PR DESCRIPTION
**Pull request by ekpyron**

### CODE REVIEW

**Comment by ekpyron** [comment]
Note that I did not switch to dynamic linking to boost for now - I think we should do that for the non-static builds, but probably better to do one change at a time.

Also: We can consider building a static z3 and linking against it in the future (somewhat related to that: #5376).

File: scripts/release_ppa.sh [hunk 1]
```diff
@@ -41,33 +47,35 @@ else
...
-    pparepo=ethereum
-    ppafilesurl=https://launchpad.net/~ethereum/+archive/ubuntu/ethereum/+files
-fi
-
 keyid=70D110489D66E2F6
 email=builds@ethereum.org
 packagename=solc
 
-for distribution in trusty xenial bionic cosmic
+for distribution in xenial bionic cosmic static
```
Comment (chriseth) [inline scripts/release_ppa.sh]: Perhaps `STATIC` to make it stand out?

File: scripts/release_ppa.sh [hunk 2]
```diff
@@ -41,33 +47,35 @@ else
...
     Z3DEPENDENCY=""
+    CMAKE_OPTIONS="-DSOLC_LINK_STATIC=On"
 else
+    if [ "$branch" = develop ]
+    then
+        pparepo=ethereum-dev
+        ppafilesurl=https://launchpad.net/~ethereum/+archive/ubuntu/ethereum-dev/+files
+    else
+        pparepo=ethereum
+        ppafilesurl=https://launchpad.net/~ethereum/+archive/ubuntu/ethereum/+files
```
Comment (chriseth) [inline scripts/release_ppa.sh]: Can't we derive that from `$pparepo` for all three cases?

File: scripts/release_ppa.sh [hunk 3]
```diff
@@ -224,7 +232,12 @@ EMAIL="$email" dch -v 1:${debversion}-${versionsuffix} "git build of ${commithas
 debuild -S -d -sa -us -uc
 
 # prepare .changes file for Launchpad
-sed -i -e s/UNRELEASED/${distribution}/ -e s/urgency=medium/urgency=low/ ../*.changes
+if [ $distribution = static ]
+then
+    sed -i -e s/UNRELEASED/cosmic/ -e s/urgency=medium/urgency=low/ ../*.changes
```
Comment (chriseth) [inline scripts/release_ppa.sh]: cosmic could be a parameter set at the beginning of the script

File: scripts/release_ppa.sh (hunk 2 above)
Comment (ekpyron) [reply scripts/release_ppa.sh]: ~~Used for the ``dput`` call at the very end.~~ sorry - misread.

Comment (ekpyron) [reply scripts/release_ppa.sh]: We indeed can - done.

**Comment by ekpyron** [comment]
@chriseth Updated.

Once we think the script is fine we can just try to run it before merging the PR and check whether it works.

We have to modify the release checklist as well (should I change it in this PR already?).

**Comment by chriseth** [comment]
Yes, please update the release checklist. We might also want to mention that in the changelog.

**Comment by ekpyron** [comment]
Actually the ``versionsuffix`` variable should probably have used a lower-case ``static`` and not ``${distribution}``, which now is ``STATIC`` - but it seems it will be converted to lowercase automatically... Should I change it anyways?

**Comment by stackenbotten** [comment]
There was an error when running `test_check_style` for commit `49a8c88f4ac5ee08a91585820b33fe7874405144`:
```
Error: Trailing whitespace found:
 ReleaseChecklist.md:13: - [ ] Once the ``~ethereum/ubuntu/ethereum-static`` PPA build is finished, copy the static package to the ``~ethereum/ubuntu/ethereum`` PPA for the destination series ``Trusty`` while selecting ``Copy existing binaries``. 

```
Please check that your changes are working as intended.

**Review by chriseth — APPROVED** [review]

**Pull request merged**